File: javascript/semana9/Persona.js
class Persona{

    static contadorPersonas = 0;

    constructor(nombre, apellido, edad){
        this._idPersona = ++Persona.contadorPersona;
        this._nombre = nombre;
        this._apellido = apellido;
        this._edad = edad;
    }

    get idPersona(){
        return this._idPersona;
    }

    get nombre(){
        return this._nombre;
    }

    set nombre(nombre){
        this._nombre = nombre;
    }

    get apellido(){
        return this._apellido;
    }

    set apellido(apellido){
        return this._apellido;
    }

    get edad(){
        return this.edad;
    }

    set edad(edad){
        return this.edad;
    }

    toString(){
        return this._idPersona+' '+this._nombre+' '+this._apellido+' '+this._edad;
    }
}

File: java/Semana2/Persona.java
package domain;
    
public class Persona {
    private final int idPersona;
    private static int contadorPersonas;
    
    static{ //Bloque de inicialización estático
        System.out.println("Ejecución del bloque estático");
        ++Persona.contadorPersonas;
        //idPersona = 10; No es estático por eso tenemos un error
    }
    
    { //Bloque de inicialización NO estático (contexto dinámico)
        System.out.println("Ejecución de bloque NO estático");
        this.idPersona = Persona.contadorPersonas++;//Incrementamos el atributo
        
    }
    //Los bloques de inicializacíon se ejecutan antes del constructor
    
    public Persona(){
        System.out.println("Ejecucuion del constructor");
    }
    
    public int getidPersona(){
        return this.idPersona;
    }

    @Override
    public String toString() {
        return "Persona{" + "idPersona=" + idPersona + '}';
    }
    
    
}
